src/Pages/Logout.js

```diff
@@ -1,13 +1,33 @@
+import {useEffect} from 'react';
 import {useDispatch, useSelector} from 'react-redux';
+import {useNavigate} from 'react-router-dom';
+import {toast} from 'react-toastify';
 import {removeUser} from '../redux/redux-reducers/userstore';
 
 const Logout = () => {
+  const navigate = useNavigate();
   const loggedIn = useSelector((state) => state.userStore.loggedIn);
   const dispatch = useDispatch();
-  if (loggedIn) {
-    dispatch(removeUser());
-    window.location.reload();
-  } else return <h1 style={{color: 'white', fontFamily: 'monospace'}}>You have been logged out, thank you for using our movieApp</h1>;
+
+  dispatch(removeUser());
+  useEffect(() => {
+    const logoutWait = new Promise((resolve) => {
+      setTimeout(() => {
+        resolve();
+        navigate({
```

Comment:
Idea looks cool, we might convert this to a bit cleaner way.

Example: 
we can have navigation to login as a function ``` const loginMock = () => {
setTimeout(navigate({pathename: '/login'}),1500)} ``` we call it mock since it mimics actual API call 
than in the setTimeot we can do something like this
``` resolve(loginMock) ```

However this is not a must, just the way we can approach this task.

Reply:
I was doing this but couldn't get it to work I know now why, I didn't put the function in the resolve(args)